File: regions/system/library/go/scheduler-client/scheduler_client.go
package schedulerclient

import (
	"context"
	"encoding/json"
	"fmt"
	"sync"
	"time"
)

// JobStatus はジョブの状態。
type JobStatus string

const (
	JobStatusPending   JobStatus = "pending"
	JobStatusRunning   JobStatus = "running"
	JobStatusCompleted JobStatus = "completed"
	JobStatusFailed    JobStatus = "failed"
	JobStatusPaused    JobStatus = "paused"
	JobStatusCancelled JobStatus = "cancelled"
)

// Schedule はジョブのスケジュール。
type Schedule struct {
	Type     string         `json:"type"` // "cron", "one_shot", "interval"
	Cron     string         `json:"cron,omitempty"`
	OneShot  *time.Time     `json:"one_shot,omitempty"`
	Interval *time.Duration `json:"interval,omitempty"`
}

// JobRequest はジョブ登録リクエスト。
type JobRequest struct {
	Name        string          `json:"name"`
	Schedule    Schedule        `json:"schedule"`
	Payload     json.RawMessage `json:"payload"`
	MaxRetries  uint32          `json:"max_retries"`
	TimeoutSecs uint64          `json:"timeout_secs"`
}

// Job はジョブ情報。
type Job struct {
	ID          string          `json:"id"`
	Name        string          `json:"name"`
	Schedule    Schedule        `json:"schedule"`
	Status      JobStatus       `json:"status"`
	Payload     json.RawMessage `json:"payload"`
	MaxRetries  uint32          `json:"max_retries"`
	TimeoutSecs uint64          `json:"timeout_secs"`
	CreatedAt   time.Time       `json:"created_at"`
	NextRunAt   *time.Time      `json:"next_run_at,omitempty"`
}

// JobFilter はジョブ一覧取得フィルター。
type JobFilter struct {
	Status     *JobStatus `json:"status,omitempty"`
	NamePrefix *string    `json:"name_prefix,omitempty"`
}

// JobExecution は実行履歴。
type JobExecution struct {
	ID         string     `json:"id"`
	JobID      string     `json:"job_id"`
	StartedAt  time.Time  `json:"started_at"`
	FinishedAt *time.Time `json:"finished_at,omitempty"`
	Result     string     `json:"result"`
	Error      string     `json:"error,omitempty"`
}

// JobCompletedEvent はジョブ完了イベント。
type JobCompletedEvent struct {
	JobID       string    `json:"job_id"`
	ExecutionID string    `json:"execution_id"`
	CompletedAt time.Time `json:"completed_at"`
	Result      string    `json:"result"`
}

// SchedulerClient はスケジューラークライアントのインターフェース。
type SchedulerClient interface {
	CreateJob(ctx context.Context, req JobRequest) (Job, error)
	CancelJob(ctx context.Context, jobID string) error
	PauseJob(ctx context.Context, jobID string) error
	ResumeJob(ctx context.Context, jobID string) error
	GetJob(ctx context.Context, jobID string) (Job, error)
	ListJobs(ctx context.Context, filter JobFilter) ([]Job, error)
	GetExecutions(ctx context.Context, jobID string) ([]JobExecution, error)
}

// InMemoryClient はメモリ内のスケジューラークライアント。
type InMemoryClient struct {
	mu   sync.Mutex
	jobs map[string]Job
	seq  int
}

// NewInMemoryClient は新しい InMemoryClient を生成する。
func NewInMemoryClient() *InMemoryClient {
	return &InMemoryClient{
		jobs: make(map[string]Job),
	}
}

func (c *InMemoryClient) CreateJob(_ context.Context, req JobRequest) (Job, error) {
	c.mu.Lock()
	defer c.mu.Unlock()
	c.seq++
	job := Job{
		ID:          fmt.Sprintf("job-%03d", c.seq),
		Name:        req.Name,
		Schedule:    req.Schedule,
		Status:      JobStatusPending,
		Payload:     req.Payload,
		MaxRetries:  req.MaxRetries,
		TimeoutSecs: req.TimeoutSecs,
		CreatedAt:   time.Now(),
	}
	c.jobs[job.ID] = job
	return job, nil
}

func (c *InMemoryClient) CancelJob(_ context.Context, jobID string) error {
	c.mu.Lock()
	defer c.mu.Unlock()
	job, ok := c.jobs[jobID]
	if !ok {
		return fmt.Errorf("job not found: %s", jobID)
	}
	job.Status = JobStatusCancelled
	c.jobs[jobID] = job
	return nil
}

func (c *InMemoryClient) PauseJob(_ context.Context, jobID string) error {
	c.mu.Lock()
	defer c.mu.Unlock()
	job, ok := c.jobs[jobID]
	if !ok {
		return fmt.Errorf("job not found: %s", jobID)
	}
	job.Status = JobStatusPaused
	c.jobs[jobID] = job
	return nil
}

func (c *InMemoryClient) ResumeJob(_ context.Context, jobID string) error {
	c.mu.Lock()
	defer c.mu.Unlock()
	job, ok := c.jobs[jobID]
	if !ok {
		return fmt.Errorf("job not found: %s", jobID)
	}
	job.Status = JobStatusPending
	c.jobs[jobID] = job
	return nil
}

func (c *InMemoryClient) GetJob(_ context.Context, jobID string) (Job, error) {
	c.mu.Lock()
	defer c.mu.Unlock()
	job, ok := c.jobs[jobID]
	if !ok {
		return Job{}, fmt.Errorf("job not found: %s", jobID)
	}
	return job, nil
}

func (c *InMemoryClient) ListJobs(_ context.Context, filter JobFilter) ([]Job, error) {
	c.mu.Lock()
	defer c.mu.Unlock()
	var result []Job
	for _, job := range c.jobs {
		if filter.Status != nil && job.Status != *filter.Status {
			continue
		}
		if filter.NamePrefix != nil && len(*filter.NamePrefix) > 0 {
			if len(job.Name) < len(*filter.NamePrefix) || job.Name[:len(*filter.NamePrefix)] != *filter.NamePrefix {
				continue
			}
		}
		result = append(result, job)
	}
	return result, nil
}

func (c *InMemoryClient) GetExecutions(_ context.Context, _ string) ([]JobExecution, error) {
	return []JobExecution{}, nil
}

// Jobs は登録済みジョブ一覧を返す。
func (c *InMemoryClient) Jobs() map[string]Job {
	c.mu.Lock()
	defer c.mu.Unlock()
	result := make(map[string]Job, len(c.jobs))
	for k, v := range c.jobs {
		result[k] = v
	}
	return result
}

File: regions/system/library/swift/scheduler-client/Sources/K1s0SchedulerClient/SchedulerClient.swift
import Foundation

public enum JobStatus: String, Sendable {
    case pending
    case running
    case completed
    case failed
    case paused
    case cancelled
}

public enum Schedule: Sendable {
    case cron(String)
    case oneShot(Date)
    case interval(Duration)
}

public struct JobRequest: Sendable {
    public let name: String
    public let schedule: Schedule
    public let payload: [String: String]
    public let maxRetries: UInt32
    public let timeoutSecs: UInt64

    public init(
        name: String,
        schedule: Schedule,
        payload: [String: String] = [:],
        maxRetries: UInt32 = 3,
        timeoutSecs: UInt64 = 60
    ) {
        self.name = name
        self.schedule = schedule
        self.payload = payload
        self.maxRetries = maxRetries
        self.timeoutSecs = timeoutSecs
    }
}

public struct Job: Sendable {
    public let id: String
    public let name: String
    public let schedule: Schedule
    public let status: JobStatus
    public let payload: [String: String]
    public let maxRetries: UInt32
    public let timeoutSecs: UInt64
    public let createdAt: Date
    public let nextRunAt: Date?

    public init(
        id: String,
        name: String,
        schedule: Schedule,
        status: JobStatus,
        payload: [String: String] = [:],
        maxRetries: UInt32 = 3,
        timeoutSecs: UInt64 = 60,
        createdAt: Date = Date(),
        nextRunAt: Date? = nil
    ) {
        self.id = id
        self.name = name
        self.schedule = schedule
        self.status = status
        self.payload = payload
        self.maxRetries = maxRetries
        self.timeoutSecs = timeoutSecs
        self.createdAt = createdAt
        self.nextRunAt = nextRunAt
    }
}

public struct JobFilter: Sendable {
    public let status: JobStatus?
    public let namePrefix: String?

    public init(status: JobStatus? = nil, namePrefix: String? = nil) {
        self.status = status
        self.namePrefix = namePrefix
    }
}

public struct JobExecution: Sendable {
    public let id: String
    public let jobId: String
    public let startedAt: Date
    public let finishedAt: Date?
    public let result: String
    public let error: String?

    public init(
        id: String,
        jobId: String,
        startedAt: Date,
        finishedAt: Date? = nil,
        result: String,
        error: String? = nil
    ) {
        self.id = id
        self.jobId = jobId
        self.startedAt = startedAt
        self.finishedAt = finishedAt
        self.result = result
        self.error = error
    }
}

public struct JobCompletedEvent: Sendable {
    public let jobId: String
    public let executionId: String
    public let completedAt: Date
    public let result: String

    public init(jobId: String, executionId: String, completedAt: Date, result: String) {
        self.jobId = jobId
        self.executionId = executionId
        self.completedAt = completedAt
        self.result = result
    }
}

public enum SchedulerError: Error, Sendable {
    case jobNotFound(jobId: String)
    case invalidSchedule(reason: String)
    case serverError(message: String)
    case timeout
}

public protocol SchedulerClient: Sendable {
    func createJob(_ req: JobRequest) async throws -> Job
    func cancelJob(jobId: String) async throws
    func pauseJob(jobId: String) async throws
    func resumeJob(jobId: String) async throws
    func getJob(jobId: String) async throws -> Job
    func listJobs(filter: JobFilter?) async throws -> [Job]
    func getExecutions(jobId: String) async throws -> [JobExecution]
}

public actor InMemorySchedulerClient: SchedulerClient {
    private var jobs: [String: Job] = [:]
    private var seq = 0

    public init() {}

    public func allJobs() -> [String: Job] {
        jobs
    }

    public func createJob(_ req: JobRequest) async throws -> Job {
        seq += 1
        let id = String(format: "job-%03d", seq)
        let job = Job(
            id: id,
            name: req.name,
            schedule: req.schedule,
            status: .pending,
            payload: req.payload,
            maxRetries: req.maxRetries,
            timeoutSecs: req.timeoutSecs
        )
        jobs[id] = job
        return job
    }

    public func cancelJob(jobId: String) async throws {
        guard let job = jobs[jobId] else {
            throw SchedulerError.jobNotFound(jobId: jobId)
        }
        jobs[jobId] = Job(
            id: job.id, name: job.name, schedule: job.schedule,
            status: .cancelled, payload: job.payload,
            maxRetries: job.maxRetries, timeoutSecs: job.timeoutSecs,
            createdAt: job.createdAt, nextRunAt: job.nextRunAt
        )
    }

    public func pauseJob(jobId: String) async throws {
        guard let job = jobs[jobId] else {
            throw SchedulerError.jobNotFound(jobId: jobId)
        }
        jobs[jobId] = Job(
            id: job.id, name: job.name, schedule: job.schedule,
            status: .paused, payload: job.payload,
            maxRetries: job.maxRetries, timeoutSecs: job.timeoutSecs,
            createdAt: job.createdAt, nextRunAt: job.nextRunAt
        )
    }

    public func resumeJob(jobId: String) async throws {
        guard let job = jobs[jobId] else {
            throw SchedulerError.jobNotFound(jobId: jobId)
        }
        jobs[jobId] = Job(
            id: job.id, name: job.name, schedule: job.schedule,
            status: .pending, payload: job.payload,
            maxRetries: job.maxRetries, timeoutSecs: job.timeoutSecs,
            createdAt: job.createdAt, nextRunAt: job.nextRunAt
        )
    }

    public func getJob(jobId: String) async throws -> Job {
        guard let job = jobs[jobId] else {
            throw SchedulerError.jobNotFound(jobId: jobId)
        }
        return job
    }

    public func listJobs(filter: JobFilter?) async throws -> [Job] {
        var result = Array(jobs.values)
        if let status = filter?.status {
            result = result.filter { $0.status == status }
        }
        if let prefix = filter?.namePrefix, !prefix.isEmpty {
            result = result.filter { $0.name.hasPrefix(prefix) }
        }
        return result
    }

    public func getExecutions(jobId: String) async throws -> [JobExecution] {
        []
    }
}
